### PR TITLE
Be app 0166

### PR DESCRIPTION
**Pull request by habibdevsy**
- order logs

### CODE REVIEW

**Review by ralsalloum — APPROVED** [review]

File: backend/src/Entity/OrderLogsEntity.php [hunk 1]
```diff
@@ -0,0 +1,83 @@
+<?php
+
+namespace App\Entity;
+
+use App\Repository\OrderLogsEntityRepository;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+
+#[ORM\Entity(repositoryClass: OrderLogsEntityRepository::class)]
+class OrderLogsEntity
```
Comment (Kenan-Hussein) [inline backend/src/Entity/OrderLogsEntity.php:10]: order status is missing

**Review by Kenan-Hussein — CHANGES_REQUESTED** [review]
Request changes

File: backend/src/Entity/CaptainEntity.php [hunk 2]
```diff
@@ -301,4 +305,34 @@ public function removeOrderChatRoomEntity(OrderChatRoomEntity $orderChatRoomEnti
...
+     * @return Collection|OrderLogsEntity[]
+     */
+    public function getOrderLogsEntity(): Collection
+    {
+        return $this->OrderLogsEntity;
+    }
+
+    public function addOrderLogsEntity(OrderLogsEntity $orderLogsEntity): self
+    {
+        if (!$this->OrderLogsEntity->contains($orderLogsEntity)) {
```
Comment (Kenan-Hussein) [inline backend/src/Entity/CaptainEntity.php:319]: to make PHP stan happy please add space after "!"

File: backend/src/Entity/OrderLogsEntity.php [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    #[ORM\GeneratedValue]
+    #[ORM\Column(type: 'integer')]
+    private $id;
+
+    #[ORM\ManyToOne(targetEntity: OrderEntity::class, inversedBy: 'OrderLogsEntity')]
+    #[ORM\JoinColumn(nullable: false)]
+    private $orderId;
+
+    #[ORM\ManyToOne(targetEntity: CaptainEntity::class, inversedBy: 'OrderLogsEntity')]
+    private $captainProfile;
```
Comment (Kenan-Hussein) [inline backend/src/Entity/OrderLogsEntity.php:23]: if this is not nullable, how you will be able to insert the first status of order: "order created" ?

File: backend/src/Entity/OrderLogsEntity.php (hunk 1 above)
Comment (Kenan-Hussein) [inline backend/src/Entity/OrderLogsEntity.php:10]: let us add branch Id also, will be much use of data

File: backend/src/Manager/Order/OrderManager.php [hunk 4]
```diff
@@ -19,6 +19,7 @@
 use App\Manager\Captain\CaptainManager;
 use DateTime;
 use App\Request\Order\OrderUpdateCaptainOrderCostRequest;
+use App\Manager\OrderLogs\OrderLogsManager;
 
 class OrderManager
```
Comment (Kenan-Hussein) [inline backend/src/Manager/Order/OrderManager.php]: Ehm Ehm:
please move this logic "create logs" to its own service.
then call it from order service.
this approach kills the "single responsibility" principle

File: backend/src/Manager/OrderLogs/OrderLogsManager.php [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(AutoMapping $autoMapping, EntityManagerInterface $entityManager, OrderLogsEntityRepository $orderLogsEntityRepository)
+    {
+        $this->autoMapping = $autoMapping;
+        $this->entityManager = $entityManager;
+        $this->orderLogsEntityRepository = $orderLogsEntityRepository;
+    }
+
+    public function create(OrderLogsCreateRequest $request): ?OrderLogsEntity
```
Comment (Kenan-Hussein) [inline backend/src/Manager/OrderLogs/OrderLogsManager.php]: what is the use case of this? please message me for this topic

File: backend/src/Request/OrderLogs/OrderLogsCreateRequest.php [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+{
+   private OrderEntity $orderId;
+
+   private StoreOwnerProfileEntity $storeOwnerProfile;
+   
+   private null|CaptainEntity $captainProfile;
+
+   private string $orderState;
+
+   private null|StoreOwnerBranchEntity $storeOwnerBranch;
```
Comment (ralsalloum) [inline backend/src/Request/OrderLogs/OrderLogsCreateRequest.php]: Why store branch may have null type while it is for sure there will be a branch to request an order from?

File: backend/src/Request/OrderLogs/OrderLogsCreateRequest.php [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+
+   /**
+    * Get the value of orderId
+    */ 
+   public function getOrderId()
```
Comment (ralsalloum) [inline backend/src/Request/OrderLogs/OrderLogsCreateRequest.php:25]: LOL, functions' return types are missing :)

File: backend/src/Request/OrderLogs/OrderLogsCreateRequest.php [hunk 8]
```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace App\Request\Rate;
```
Comment (ralsalloum) [inline backend/src/Request/OrderLogs/OrderLogsCreateRequest.php]: DANGER: namespace path is not correct.

File: backend/src/Manager/OrderLogs/OrderLogsManager.php [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+class OrderLogsManager
+{
+    private AutoMapping $autoMapping;
+    private EntityManagerInterface $entityManager;
+
+    public function __construct(AutoMapping $autoMapping, EntityManagerInterface $entityManager, OrderLogsEntityRepository $orderLogsEntityRepository)
+    {
+        $this->autoMapping = $autoMapping;
+        $this->entityManager = $entityManager;
+        $this->orderLogsEntityRepository = $orderLogsEntityRepository;
```
Comment (ralsalloum) [inline backend/src/Manager/OrderLogs/OrderLogsManager.php:21]: orderLogsEntityRepository variable is not defined.

File: backend/src/Manager/Order/StoreOrderDetailsManager.php [hunk 10]
```diff
@@ -63,4 +64,9 @@ public function createImage(string $image, int $id): ImageEntity
         return $this->imageManager->create($request);
         return $request;
     }
+
+    public function getOrderDetailsByOrderId($orderId): StoreOrderDetailsEntity
```
Comment (ralsalloum) [inline backend/src/Manager/Order/StoreOrderDetailsManager.php]: type hint is missing.

File: backend/src/Manager/Order/StoreOrderDetailsManager.php [hunk 11]
```diff
@@ -21,7 +21,8 @@ class StoreOrderDetailsManager
 
     public function __construct(private AutoMapping $autoMapping, private EntityManagerInterface $entityManager, private StoreOrderDetailsEntityRepository $storeOrderDetailsEntityRepository, private StoreOwnerBranchManager $storeOwnerBranchManager, ImageManager $imageManager)
     {
-        $this->imageManager =  $imageManager;
+        $this->imageManager = $imageManager;
+        $this->storeOrderDetailsEntityRepository = $storeOrderDetailsEntityRepository;
```
Comment (ralsalloum) [inline backend/src/Manager/Order/StoreOrderDetailsManager.php:26]: storeOrderDetailsEntityRepository variable is not defined to initialize it + you have already initialize and defined it in the constructor function definition.

File: backend/src/Service/OrderLogs/OrderLogsService.php [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    public function createOrderLogs(OrderLogsCreateRequest $request): OrderLogsResponse
+    {
+       $orderLog = $this->orderLogsManager->createOrderLogs($request);
+    
+       return $this->autoMapping->map(OrderLogsEntity::class, OrderLogsResponse::class, $orderLog);
+    }
+    
+    public function createOrderLogsRequest(OrderEntity $order): ?OrderLogsResponse
```
Comment (ralsalloum) [inline backend/src/Service/OrderLogs/OrderLogsService.php]: How could function return null as long as it returns the response of _createOrderLogs_ functions, which is **OrderLogsResponse** only?

**Review by ralsalloum — CHANGES_REQUESTED** [review]
Please check the new notes. Besides, how about adding an API to get order logs?

**Review by ralsalloum — APPROVED** [review]

**Review by Kenan-Hussein — APPROVED** [review]

**Pull request merged**